mne/decoding/search_light.py

```diff
@@ -76,13 +77,16 @@ def fit(self, X, y, **fit_params):
             _sl_fit, self.n_jobs, max_jobs=X.shape[-1], verbose=False)
         self.estimators_ = list()
         self.fit_params = fit_params
+
         # For fitting, the parallelization is across estimators.
-        mesg = 'Fitting %s' % (self.__class__.__name__,)
-        with ProgressBar(X.shape[-1], mesg=mesg, position=self.position) as pb:
+        context = _create_progressbar_context(self, X, 'Fitting')
+        with context as pb:
             estimators = parallel(
-                p_func(self.base_estimator, split, y, pb.subset(pb_idx),
+                p_func(self.base_estimator, split, y,
+                       pb.subset(pb_idx) if self.verbose else None,
```

Comment:
My one remaining comment is that, everywhere else in MNE, `verbose=<str>` can be used, so it would be nice to support it here in addition to pure True and False. For example, a user setting `verbose = 'warning'` at the top of a script then passing it to all MNE calls wouldn't have the effect they rightly expect in this PR iteration.

If think you could use (or refactor) some private function in `mne/utils` that already converts `True, False, 'info', 'debug', 'error', 'warning'` to a logging level, then use `converted_level <= logging.INFO` to be your effective boolean value

Reply:
Thanks @larsoner, I'll do that.

Reply:
Ok, you can take a look @larsoner. I now noticed that there is also `_verbose_safe_false` function in `mne.utils._logging` so I could used that and revert the `_parse_verbose` and `_check_verbose`.

Reply:
oh, no, I was wrong, `_verbose_safe_false` does something a bit different.

Reply:
@larsoner I've added `which_tqdm` argument to `Progressbar` (previously it only used environment variable for setting tqdm). It should be better now. :)

Reply:
Oh, wow, it seems that my comment is in a tangled state, and changing it in a different comment thread changed it also here!
So, let me try agin:
I added `which_tqdm` argument to `Progressbar`, take a look if this is ok.